xla/service/gpu/fusions/transpose_mlir.cc

```diff
@@ -75,7 +75,52 @@ using mlir_converter::ApplyAffineMap;
 using mlir_converter::CallTargetProvider;
 using mlir_converter::PartitionedComputation;
 
-Tiling ComputeTransposeTiling(const TransposeDescription& tiled_transpose) {
+constexpr int kPreTransposeMinorDim = 2;
+constexpr int kPreTransposeVectorizedDim = 3;
+
+int GetVectorSize(const HloFusionAnalysis& analysis,
+                  absl::InlinedVector<int64_t, 4> input_dims,
+                  Vector3 permutation, int64_t threads_per_block) {
+  int vector_size = 2;
+  int64_t threads_max = analysis.device_info().core_count() *
+                        analysis.device_info().threads_per_core_limit();
+
+  auto threads_counts = [&](int vec_size) {
+    int64_t blocks = 1;
+    for (int i = 0; i < permutation.size(); ++i) {
```

Comment:
This loop is very confusing. Why not just create a `Tiling` and ask for the actual thread counts?